arviz/tests/base_tests/test_plots_matplotlib.py

```diff
@@ -179,33 +181,33 @@ def test_plot_trace_legend(compact, combined):
     axes = plot_trace(
         idata, var_names=["home", "atts_star"], compact=compact, combined=combined, legend=True
     )
-    assert axes[0, 0].get_legend()
-    compact_legend = axes[1, 0].get_legend()
+    assert axes.get_legend()
+    compact_legend = axes.get_legend()
     if compact:
-        assert axes.shape == (2, 2)
+        assert len(axes.figure.axes) == 4
         assert compact_legend
     else:
-        assert axes.shape == (7, 2)
+        assert len(axes.figure.axes) == 14
         assert not compact_legend
 
 
 def test_plot_trace_discrete(discrete_model):
     axes = plot_trace(discrete_model)
-    assert axes.shape
+    assert axes.figure
 
 
 def test_plot_trace_max_subplots_warning(models):
     with pytest.warns(UserWarning):
         with rc_context(rc={"plot.max_subplots": 6}):
             axes = plot_trace(models.model_1)
-    assert axes.shape == (3, 2)
+    assert len(axes.figure.axes) == 6
 
 
 @pytest.mark.parametrize("kwargs", [{"var_names": ["mu", "tau"], "lines": [("hey", {}, [1])]}])
 def test_plot_trace_invalid_varname_warning(models, kwargs):
     with pytest.warns(UserWarning, match="valid var.+should be provided"):
         axes = plot_trace(models.model_1, **kwargs)
-    assert axes.shape
+    assert axes.figure
```

Comment:
why it is better to check `axes.figure` instead of just `axes`?

Reply:
why would that work? axes is numpy array?

But, true `ax.figure` kind of test that mpl axis is created.

Reply:
Hi, axes is not an np.array anymore. It is a matplotlib AxesSubplot object.

Reply:
oh yes, this is traceplot 👍